tests/integration/pilot/multiplecontrolplanes/main_test.go

```diff
@@ -0,0 +1,264 @@
+	// echo1NS is under userGroup1NS controlplane and echo2NS and echo3NS are under userGroup2NS controlplane
+	echo1NS namespace.Instance
+	echo2NS namespace.Instance
+	echo3NS namespace.Instance
+	apps    deployment.Echos
+)
+
+// TestMain defines the entrypoint for multiple controlplane tests using revisions and discoverySelectors.
+func TestMain(m *testing.M) {
+	// nolint: staticcheck
+	framework.
+		NewSuite(m).
+		SkipExternalControlPlaneTopology().
+		// Requires two CPs with specific names to be configured.
+		Label(label.CustomSetup).
+		SetupParallel(
+			namespace.Setup(&userGroup1NS, namespace.Config{Prefix: "usergroup-1", Labels: map[string]string{"usergroup": "usergroup-1"}}),
+			namespace.Setup(&userGroup2NS, namespace.Config{Prefix: "usergroup-2", Labels: map[string]string{"usergroup": "usergroup-2"}})).
+		Setup(istio.Setup(nil, func(ctx resource.Context, cfg *istio.Config) {
+			s := ctx.Settings()
+			// TODO test framework has to be enhanced to use istioNamespace in istioctl commands used for VM config
+			s.SkipWorkloadClasses = append(s.SkipWorkloadClasses, echo.VM)
+
+			cfg.Values["global.istioNamespace"] = userGroup1NS.Name()
+			cfg.SystemNamespace = userGroup1NS.Name()
+			cfg.ControlPlaneValues = fmt.Sprintf(`
+namespace: %s
+revision: usergroup-1
+meshConfig:
+  # REGISTRY_ONLY is used to verify custom resources scoping
+  outboundTrafficPolicy:
+    mode: REGISTRY_ONLY
+  # CR scoping requires discoverySelectors to be configured
+  discoverySelectors:
+    - matchLabels:
+        usergroup: usergroup-1
+values:
+  global:
+    istioNamespace: %s
+  pilot:
+    env:
+      # CR scoping is guarded by the feature flag
+      ENABLE_ENHANCED_RESOURCE_SCOPING: true`,
+				userGroup1NS.Name(), userGroup1NS.Name())
+		})).
+		Setup(istio.Setup(nil, func(ctx resource.Context, cfg *istio.Config) {
+			s := ctx.Settings()
+			// TODO test framework has to be enhanced to use istioNamespace in istioctl commands used for VM config
+			s.SkipWorkloadClasses = append(s.SkipWorkloadClasses, echo.VM)
+
+			cfg.Values["global.istioNamespace"] = userGroup2NS.Name()
+			cfg.SystemNamespace = userGroup2NS.Name()
+			cfg.ControlPlaneValues = fmt.Sprintf(`
+namespace: %s
+revision: usergroup-2
+meshConfig:
+  # REGISTRY_ONLY is used to verify custom resources scoping
+  outboundTrafficPolicy:
+    mode: REGISTRY_ONLY
+  # CR scoping requires discoverySelectors to be configured
+  discoverySelectors:
+    - matchLabels:
+        usergroup: usergroup-2
+values:
+  global:
+    istioNamespace: %s
+  pilot:
+    env:
+      # CR scoping is guarded by the feature flag
+      ENABLE_ENHANCED_RESOURCE_SCOPING: true`,
+				userGroup2NS.Name(), userGroup2NS.Name())
+		})).
+		SetupParallel(
+			// application namespaces are labeled according to the required control plane ownership.
+			namespace.Setup(&echo1NS, namespace.Config{Prefix: "echo1", Inject: true, Revision: "usergroup-1", Labels: map[string]string{"usergroup": "usergroup-1"}}),
```

Comment:
nit: I think this deploys like 10 deploments * 3 namespaces, which is really excessive.

Only takes 30s though and I don't know how to easily reduce it in the framework, so not a blocker